test/functional/logos_feature_bip341_sighash.py

```diff
@@ -0,0 +1,386 @@
+            for _ in test_case['sig_hash_types']:
+                tx.vin.append(CTxIn(COutPoint(int(tx_fan_out.hash, 16), utxo_idx), CScript()))
+                utxo_idx += 1
+            unsigned_tx = tx.serialize().hex()
+            private_keys_wif = []
+            sign_inputs = []
+            for i, spent_output in enumerate(spent_outputs):
+                sign_inputs.append({
+                    'txid': tx_fan_out.hash,
+                    'vout': key_idx + i,
+                    'amount': Decimal(spent_output.nValue) / COIN,
+                    'scriptPubKey': spent_output.scriptPubKey.hex(),
+                })
+            for i, sig_hash_type in enumerate(test_case['sig_hash_types']):
+                sighash = SignatureHashBIP341(
+                    tx_to=tx,
+                    spent_utxos=spent_outputs,
+                    sig_hash_type=sig_hash_type,
+                    input_index=i,
+                    executed_script_hash=hashlib.sha256(executed_scripts[key_idx]).digest(),
+                    codeseparator_pos=test_case.get('codesep', 0xffff_ffff),
+                )
+                if test_case.get('schnorr', False):
+                    signature = private_keys[key_idx].sign_schnorr(sighash)
+                else:
+                    signature = private_keys[key_idx].sign_ecdsa(sighash)
+                signature += bytes([test_case.get('suffix', sig_hash_type & 0xff)])
+                if test_case.get('opcodes'):
+                    tx.vin[i].scriptSig = CScript([signature, executed_scripts[key_idx]])
+                elif test_case.get('is_p2pk'):
+                    tx.vin[i].scriptSig = CScript([signature])
+                else:
+                    tx.vin[i].scriptSig = CScript([signature, public_keys[key_idx].get_bytes()])
+
+                sig_hash_type_str = self.get_sig_hash_type_str(sig_hash_type)
+                if sig_hash_type_str is not None and 'opcodes' not in test_case and 'error' not in test_case:
+                    private_key_wif = bytes_to_wif(private_keys[key_idx].get_bytes())
+                    raw_tx_signed = self.nodes[0].signrawtransactionwithkey(
+                        unsigned_tx, [private_key_wif], sign_inputs, sig_hash_type_str)['hex']
+                    signed_tx = CTransaction()
+                    signed_tx.deserialize(io.BytesIO(bytes.fromhex(raw_tx_signed)))
+                    sig = list(CScript(signed_tx.vin[i].scriptSig))[0]
+                    pubkey = private_keys[key_idx].get_pubkey()
+                    sighash = SignatureHashBIP341(
+                        tx_to=tx,
+                        spent_utxos=spent_outputs,
+                        sig_hash_type=sig_hash_type & 0xff,
+                        input_index=i,
+                        executed_script_hash=hashlib.sha256(executed_scripts[key_idx]).digest(),
+                    )
+                    assert pubkey.verify_ecdsa(sig[:-1], sighash)
```

Comment:
This verifies the signature provided by `signrawstransaction` ? Maybe add a comment.

Reply:
Will fix